src/Domain/AreaElement.hpp

```diff
@@ -0,0 +1,122 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Utilities/Gsl.hpp"
+
+/*!
+ * \brief Compute the Euclidean area element from the inverse jacobian.
+ *
+ * \details The Euclidean area element of a surface \f$\Sigma\f$ with
+ * constant logical coordinate \f$\xi^i\f$ is given by:
+ *
+ * \f{equation}
+ * J^\Sigma = J \sqrt{\delta^{jk} (J^{-1})^i_j (J^{-1})^i_k}
+ * \f}
+ *
+ * where \f$J^i_j = \partial x^i / \xi^j\f$ is the volume Jacobian with
+ * determinant \f$J\f$ and inverse \f$(J^{-1})^i_j = \partial \xi^i / \partial
+ * x^j\f$. The determinant of the inverse Jacobian can be passed as an argument
+ * as an overload, otherwise it will be calculated.
+ *
+ * \note The curved space area element as well as time dependent maps are not
```

Comment:
Maybe I am confused, but isn't `area_element` precisely the curved space area element?

Reply:
yes, well spotted